pallets/pallet-did-lookup/src/lib.rs

```diff
@@ -306,22 +376,22 @@ pub mod pallet {
 			// *** NO FAILURE beyond the reserve call ***
 			CurrencyOf::<T>::reserve(&record.deposit.owner, record.deposit.amount)?;
 
-			ConnectedDids::<T>::mutate(&account, |did_entry| {
+			ConnectedDidsV2::<T>::mutate(&account, |did_entry| {
 				if let Some(old_connection) = did_entry.replace(record) {
-					ConnectedAccounts::<T>::remove(&old_connection.did, &account);
+					ConnectedAccountsV2::<T>::remove(&old_connection.did, &account);
 					Self::deposit_event(Event::<T>::AssociationRemoved(account.clone(), old_connection.did));
 					kilt_support::free_deposit::<AccountIdOf<T>, CurrencyOf<T>>(&old_connection.deposit);
 				}
 			});
-			ConnectedAccounts::<T>::insert(&did_identifier, &account, ());
+			ConnectedAccountsV2::<T>::insert(&did_identifier, &account, ());
 			Self::deposit_event(Event::AssociationEstablished(account, did_identifier));
 
 			Ok(())
 		}
 
-		pub(crate) fn remove_association(account: AccountIdOf<T>) -> DispatchResult {
-			if let Some(connection) = ConnectedDids::<T>::take(&account) {
-				ConnectedAccounts::<T>::remove(&connection.did, &account);
+		pub(crate) fn remove_association(account: LinkableAccountIdOf<T>) -> DispatchResult {
```

Comment:
Similar as above: in this case it might be that a removal will fail if the account has not been migrated yet. It probably deserves a comment until the next runtime.